### PR TITLE
[RFC] cacheline concurrency changes for parallel eviction

### PR DESCRIPTION
**Pull request by arutk**

### CODE REVIEW

File: src/engine/cache_engine.h [hunk 1]
```diff
@@ -12,6 +12,7 @@ struct ocf_request;
 #define LOOKUP_HIT 5
 #define LOOKUP_MISS 6
 #define LOOKUP_INSERTED 8
+#define LOOKUP_REMAPPED 9
```
Comment (mmichal10) [inline src/engine/cache_engine.h:15]: You mentioned `LOOKUP_REPLACED` in commit message

File: src/concurrency/ocf_cache_line_concurrency.c [hunk 2]
```diff
@@ -756,6 +764,11 @@ static int _ocf_req_trylock_rd(struct ocf_request *req)
 	if (ret == OCF_LOCK_NOT_ACQUIRED) {
 		/* Request is not locked, discard acquired locks */
 		for (; i >= 0; i--) {
+			if (!_ocf_req_needs_cl_lock(req, i)) {
+				/* nothing to lock */
```
Comment (mmichal10) [inline src/concurrency/ocf_cache_line_concurrency.c:768]: Shouldn't the comment be sth like `no lock to discard`?

File: src/concurrency/ocf_cache_line_concurrency.c [hunk 3]
```diff
@@ -815,6 +828,9 @@ static int _ocf_req_lock_rd(struct ocf_request *req, ocf_req_async_lock_cb cb)
 
 err:
 	for (; i >= 0; i--) {
+		if (req->map[i].status <= LOOKUP_MISS) {
```
Comment (mmichal10) [inline src/concurrency/ocf_cache_line_concurrency.c:831]: Don't you think `req->map[i].status == LOOKUP_MISS || req->map[i].status == LOOKUP_HIT` would be more readable?

Comment (arutk) [reply src/concurrency/ocf_cache_line_concurrency.c:831]: this is an error, I meant to use 
if (!_ocf_req_needs_cl_lock(req, i))

**Pull request closed without merging**